process/current_drive.py

```diff
@@ -403,11 +463,100 @@ def cudriv(self, output: bool):
                     dene20 * physics_variables.rmajor
                 )
                 current_drive_variables.effcd = effrfss
-
                 # EBWs can only couple to plasma if cyclotron harmonic is above plasma density cut-off;
                 # this behaviour is captured in the following function (ref issue #1262):
                 # current_drive_variables.harnum = cyclotron harmonic number (fundamental used as default)
                 # contant 'a' controls sharpness of transition
+                a = 0.1e0
+
+                fc = (
+                    1.0e0
+                    / (2.0e0 * np.pi)
+                    * current_drive_variables.harnum
+                    * constants.echarge
+                    * physics_variables.bt
+                    / constants.emass
+                )
+                fp = (
+                    1.0e0
+                    / (2.0e0 * np.pi)
+                    * np.sqrt(
+                        physics_variables.dene
+                        * constants.echarge**2
+                        / (constants.emass * constants.epsilon0)
+                    )
+                )
+
+                density_factor = 0.5e0 * (
+                    1.0e0 + np.tanh((2.0e0 / a) * ((fp - fc) / fp - a))
+                )
+
+                current_drive_variables.effcd = (
+                    current_drive_variables.effcd * density_factor
+                )
+                effrfss = effrfss * density_factor
```

Comment:
470 to this line doesn't make sense to me. Why have we changed how the old switch behaves in this PR?

Reply:
Realised that the cutoff function section was commented out so added it back in. The cutoff function was in for the `iefrffix` but not `iefrf`